internal/handler/msg_find.go

```diff
@@ -216,6 +216,19 @@ func (h *Handler) makeFindQueryParams(params *common.FindParams, cInfo *backends
 	case params.Sort.Len() == 0 && cInfo.Capped():
 		// Pushdown default recordID sorting for capped collections
 		qp.Sort = must.NotFail(types.NewDocument("$natural", int64(1)))
+	case params.Sort.Len() == 1:
+		if params.Sort.Keys()[0] != "$natural" {
+			break
+		}
+
+		if !cInfo.Capped() {
+			return nil, handlererrors.NewCommandErrorMsgWithArgument(
+				handlererrors.ErrNotImplemented,
+				"$natural sort for non-capped collection is not supported.",
+				"find")
+		}
```

Comment:
Left `NotImplemented` here, as apparently it's possible to run `$natural` against non-capped collection on mongo

Reply:
It makes sense in some use cases, but yeah, let's not support it for now